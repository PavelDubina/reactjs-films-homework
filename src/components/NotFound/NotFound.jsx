/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React from 'react'
import { Link } from 'react-router-dom'
import styles from './NotFound.scss'

export const NotFound = () => (
  <div className={styles.container}>
    <span className={styles.error}>Error 404</span>
    <span className={styles.title}>Page not found</span>
    <Link className={styles.link} to="/" data-testid="not-found-link">
      Go Back
    </Link>
  </div>
)
