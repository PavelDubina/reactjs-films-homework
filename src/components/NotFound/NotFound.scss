/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

.container {
  flex-grow: 1;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  height: 100%;
  width: 100%;
  font-size: 70px;
  color: #888;

  @media (max-width: 599px) {
    font-size: 50px;
  }
}

.error {
  padding: 0 15px 0 0;
  animation: type 0.5s alternate infinite;
}

.link {
  padding: 0 10px;
  margin: 10px 0 0 0;
  font-size: 50px;
  text-decoration: none;
  border: 1px solid rgb(112, 112, 112);
  border-radius: 10px;
  background-color: #888;
  color: #fff;

  &:hover {
    background-color: rgb(112, 112, 112);
  }

  @media (max-width: 599px) {
    font-size: 35px;
    padding: 0 5px;
  }
}

@keyframes type {
  from {
    box-shadow: inset -5px 0 0 #888;
  }

  to {
    box-shadow: inset -5px 0 0 transparent;
  }
}
