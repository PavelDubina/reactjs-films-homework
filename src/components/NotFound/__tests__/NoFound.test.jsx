/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React from 'react'
import { create } from 'react-test-renderer' // ES6
import { BrowserRouter as Router } from 'react-router-dom'
import { NotFound } from '../NotFound'

describe('NotFound', () => {
  it('Should render and match snapshot', () => {
    const component = create(
      <Router>
        <NotFound />
      </Router>,
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
