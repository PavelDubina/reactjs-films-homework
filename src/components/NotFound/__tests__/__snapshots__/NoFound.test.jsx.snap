// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Should render and match snapshot 1`] = `
<div
  className="container"
>
  <span
    className="error"
  >
    Error 404
  </span>
  <span
    className="title"
  >
    Page not found
  </span>
  <a
    className="link"
    data-testid="not-found-link"
    href="/"
    onClick={[Function]}
  >
    Go Back
  </a>
</div>
`;
